packages/design-system/src/components/FilterTag/FilterTag.tsx

```diff
@@ -0,0 +1,104 @@
+import { ClearIcon } from '../ClearIcon';
+import React from 'react';
+import classNames from 'classnames';
+
+const actionableKeys = ['Enter', 'Backspace', 'Delete'];
+const idFromValue = (value: string) => `${value}-tag`;
+
+export interface FilterTagProps {
+  /**
+   * Id for filter tag button. If not provided, a unique id will be automatically generated and used.
+   */
+  id?: string;
+  /**
+   * Class to be applied to the outer `<div>` that contains filter tag button.
+   */
+  className?: string;
+  /**
+   * Text for the filter tag
+   */
+  label: string;
+  /**  
+   *
+   Value for filter tag 
+   */
+  value: string;
+  /**
+   *  For screenreaders, text to read for removal
+   */
+  removeText: string;
+  /**
+   * When false, filter tag is not removable
+   */
+  readOnly?: boolean;
+  /**
+   * Function to call when filter tag is dismissed, will return value
+   */
+  onClear?: (value: string) => void;
+}
+
+interface FilterTagState {
+  tagDismissed: boolean;
+}
+
+export class FilterTag extends React.Component<FilterTagProps, FilterTagState> {
+  constructor(props: FilterTagProps) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.state = {
+      tagDismissed: false,
+    };
+  }
+
+  handleClick(): void {
+    const { readOnly, onClear, value } = this.props;
+    if (!readOnly) {
+      // Call on onClear passed in as a prop to component
+      if (onClear) {
+        onClear(value);
+      }
+
+      // Set tag as dismissed so it won't display
+      this.setState({ tagDismissed: true });
+    }
+  }
+
+  handleKeyDown(evt: React.KeyboardEvent): void {
+    if (!this.props.readOnly && actionableKeys.includes(evt.key)) {
+      this.handleClick();
+      evt.preventDefault();
+    }
+  }
+
+  render(): React.ReactNode {
+    const { id, label, removeText, readOnly, className } = this.props;
+    const rootClassNames = classNames('ds-c-filter-tag', className);
+    const readOnlyTag = readOnly ? 'ds-c-filter-tag__button--read-only' : '';
+    const buttonClassNames = classNames('ds-c-filter-tag__button', readOnlyTag);
+    return (
+      <>
+        {!this.state.tagDismissed && (
+          <div className={rootClassNames}>
+            <button
```

Comment:
I'm not sure about this, just want to check with you all, is <div> tag needed here or can we just have <button> tag for this filter tag component?